Dockerfile

```diff
@@ -16,20 +16,12 @@ RUN dnf -q -y update
 
 ENV EXTERNAL_SOURCE=.
 ENV CONTAINER_SOURCE=/opt/app-root/src
-WORKDIR /workspace
+WORKDIR $CONTAINER_SOURCE
 
-# cache deps before building and copying source so that we don't need to re-download as much
-# and so that source changes don't invalidate our downloaded layer
-COPY $EXTERNAL_SOURCE/go.mod ./go.mod
-COPY $EXTERNAL_SOURCE/go.sum ./go.sum
+# Copy the entire source code into the container first then download Go module dependencies.
+COPY $EXTERNAL_SOURCE $CONTAINER_SOURCE
```

Comment:
Could you expand a little bit on the motivations behind this PR? I feel like separating the `go.mod` and `go.sum` copy from the rest of the source code helps make the most of layer caching. Otherwise, the `go mod download` layer will be invalidated whenever there is a code change, even if the dependencies are not modified.

Reply:
@rm3l We would like to integrate Sealights in the operator. When we perform Sealights scanning in Konflux, Sealights generates an package that needs to be added to the build container runtime. That is why I put COPY . . to container. Do you have a better approach on go mod download?